xbmc/utils/DisplayInfo.cpp

```diff
@@ -0,0 +1,170 @@
+/*
+ *  Copyright (C) 2023 Team Kodi
+ *  This file is part of Kodi - https://kodi.tv
+ *
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *  See LICENSES/README.md for more information.
+ */
+
+#include "DisplayInfo.h"
+
+#include "utils/log.h"
+
+extern "C"
+{
+#include <libdisplay-info/cta.h>
+#include <libdisplay-info/edid.h>
+#include <libdisplay-info/info.h>
+}
+
+using namespace KODI;
+using namespace UTILS;
+
+void CDisplayInfo::DiInfoDeleter::operator()(di_info* p)
+{
+  di_info_destroy(p);
+}
+
+std::unique_ptr<CDisplayInfo> CDisplayInfo::Create(std::vector<uint8_t> edid)
+{
+  struct DisplayInfoMaker : public CDisplayInfo
+  {
+    DisplayInfoMaker(std::vector<uint8_t> edid) : CDisplayInfo(edid) {}
+  };
+
+  auto info = std::make_unique<DisplayInfoMaker>(edid);
+  if (!info->IsValid())
+    return {};
+
+  info->Parse();
+
+  info->LogInfo();
+
+  return info;
+}
+
+CDisplayInfo::CDisplayInfo(std::vector<uint8_t> edid)
+{
+  m_info.reset(di_info_parse_edid(edid.data(), edid.size()));
+}
+
+CDisplayInfo::~CDisplayInfo() = default;
```

Comment:
`m_info` should be destroyed via `di_info_destroy` here.

Reply:
Oh, there's some C++ stuff going on with `DiInfoDeleter`. Alright, please ignore this comment then.